src/NWCClient.ts

```diff
@@ -1048,4 +1085,59 @@ export class NWCClient {
       );
     }
   }
+
+  private async _checkCompatibility() {
+    if (!this.version) {
+      const walletServiceInfo = await this.getWalletServiceInfo();
+      const compatibleVersion = this.selectHighestCompatibleVersion(
+        walletServiceInfo.versions,
+      );
+      if (!compatibleVersion) {
+        throw new Nip47UnsupportedVersionError(
+          `no compatible version found between wallet and client`,
+          "UNSUPPORTED_VERSION",
+        );
+      }
+      this.version = compatibleVersion;
+    }
+  }
+
+  private selectHighestCompatibleVersion(
```

Comment:
It is nice, but I am wondering if such a complex algorithm is needed?

What about just:
if the wallet supports version 1.0, use it. Otherwise use 0.0.

As Bumi mentioned:
> I do not expect many versions (and rather bigger changes like the switch to nip-44) because too many versions (of a spec) lead to incompatibilities which must be avoided.

Reply:
Hmmm, I was also thinking about this, but then thought to add this as this is an sdk and can face any possibility... (a weird case where the wallet service info gives "0.5" or "1.1" for example)